### PR TITLE
validators: T4875: use file-path to replace validator 'interface-name'

### PR DESCRIPTION
**Pull request by jestabro**
## Change Summary

As a subtask of [T4795: Cleanup custom python validators](https://phabricator.vyos.net/T4795), replace the Python script validator `interface-name` with a (regex || file-path check of /sys/class/net/devname). The file-path validator was introduced in T4798.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [X] Other (please describe):
         Performance optimization.

## Related Task(s)

* https://phabricator.vyos.net/T4875
* https://phabricator.vyos.net/T4798

## Checklist:

- [X] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [X] I have linked this PR to one or more Phabricator Task(s)
- [X] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [X] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Review by dmbaturin — APPROVED** [review]
I _think_ this is fine, but we need to be careful. The old logic was "any valid interface name", while the new one is "any interface that currently exists in the system".

We need to make sure that:

* The new validator isn't used in places that should admit any name that isn't invalid.
* All services that use the new validator for a good reason have higher priorities than any interface configuration scripts so that interfaces are guaranteed to exist.

For services that do not fail when made to listen on non-existent interfaces, this may also introduce a new commit failure mode. If any service would tolerate being set to listen on a non-existent interface, the old logic allowed such commits to succeed.
Not those configs will start failing because non-existent interfaces will not fail at the validation stage.

**Comment by jestabro** [comment]
The old logic in validator 'interface-name' OR's a regex with existence of interface in /sys/class/net; the new logic in
include/constraint/interface-name.xml.in does the same. One can confirm by, say, setting a not-yet-existing dummy interface as target.

File: interface-definitions/include/constraint/interface-name.xml.in [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<!-- include start from constraint/interface-name.xml.in -->
+<regex>(bond|br|dum|en|ersp|eth|gnv|lan|l2tp|l2tpeth|macsec|peth|ppp|pppoe|pptp|sstp|tun|vti|vtun|vxlan|wg|wlan|wwan)[0-9]+(.\d+)?|lo</regex>
```
Comment (sever-sever) [inline interface-definitions/include/constraint/interface-name.xml.in]: We also have virtual-ethernet `veth` interfaces.
```
set interfaces virtual-ethernet veth0 peer-name veth1
```

**Review by sever-sever — CHANGES_REQUESTED** [review]

File: interface-definitions/include/constraint/interface-name.xml.in (hunk 1 above)
Comment (jestabro) [reply interface-definitions/include/constraint/interface-name.xml.in]: Good catch, @sever-sever ; I'll update.

**Review by sever-sever — APPROVED** [review]

**Pull request merged**